### ISSUE TITLE
Deep nested types are not sent correctly by the UI.

### ISSUE THREAD

**Issue by moelsayed** [rancher/rancher#23210]
Version: current master

We observed that when using types with several nesting levels, at some point the content of the nested types are dropped.
<img width="646" alt="Screen Shot 2019-10-02 at 7 42 34 PM" src="https://user-images.githubusercontent.com/4588065/66154309-74d3fa00-e61d-11e9-942b-be4f7e6da0e2.png">
<img width="576" alt="Screen Shot 2019-10-02 at 7 44 26 PM" src="https://user-images.githubusercontent.com/4588065/66154326-7b627180-e61d-11e9-9920-0d33f55f0456.png">

**Comment by vincent99** [comment]
There's a recursion limit (of 10) so that the browser doesn't explode if there's a circular reference somewhere.  Is this a theoretical concern or you have an actual use-case where something is reasonably going to be more than 11 levels deep?  I'm not opposed to making the limit somewhat higher, but would not be a fan of removing it entirely.

**Comment by mrajashree** [comment]
@vincent99 it's for this https://github.com/rancher/types/pull/977

**Comment by westlywright** [comment]
Available in `latest-2.3`.

**Comment by sowmyav27** [comment]
**reproduced the issue on `rancher:2.3.2`**

- input yaml in cluster (rke cluster)
```
services:
    etcd:
      backup_config:
        enabled: true
        interval_hours: 12
        retention: 6
        safe_timestamp: false
        1:
          2:
            3: 
              4:
                5:
                  6:
                    7: 
                      8:
                        9:
                          10:
                            11: 12
```

**In `rancher:2.3.2`**
**Request:**

<img width="481" alt="Screen Shot 2019-11-01 at 11 32 57 PM" src="https://user-images.githubusercontent.com/26032343/68045925-aa453380-fd00-11e9-94a7-79257dc65c1e.png">

**In `2.3-head` - latest - (11/01/2019)**
<img width="679" alt="Screen Shot 2019-11-01 at 11 38 33 PM" src="https://user-images.githubusercontent.com/26032343/68045971-cc3eb600-fd00-11e9-8b9c-8fe552d6adea.png">

- Also verified the depth is increased to 20
- Input yaml:
```
services:
    etcd:
      backup_config:
        enabled: true
        interval_hours: 12
        retention: 6
        safe_timestamp: false
        1:
         2:
           3:
             4:
               5:
                 6:
                   7:
                     8:
                       9:
                         10:
                           11:
                             12:
                               13:
                                 14:
                                   15:
                                     16:
                                       17:
                                         18:
                                           19:
                                             20:
                                               21: 22
```

**Request:**
<img width="678" alt="Screen Shot 2019-11-01 at 11 23 06 PM" src="https://user-images.githubusercontent.com/26032343/68046035-e9738480-fd00-11e9-9f36-2f0cc6268892.png">
